### PR TITLE
Add Albert example

### PR DESCRIPTION
**Pull request by kwen2501**
## Description

Add Albert example.

Model description: https://huggingface.co/docs/transformers/model_doc/albert

Model architecture:
```
AlbertForMaskedLM(
  (albert): AlbertModel(
    (embeddings): AlbertEmbeddings(
      (word_embeddings): Embedding(30000, 128, padding_idx=0)
      (position_embeddings): Embedding(512, 128)
      (token_type_embeddings): Embedding(2, 128)
      (LayerNorm): LayerNorm((128,), eps=1e-12, elementwise_affine=True)
      (dropout): Dropout(p=0, inplace=False)
    )
    (encoder): AlbertTransformer(
      (embedding_hidden_mapping_in): Linear(in_features=128, out_features=4096, bias=True)
      (albert_layer_groups): ModuleList(
        (0): AlbertLayerGroup(
          (albert_layers): ModuleList(
            (0): AlbertLayer(
              (full_layer_layer_norm): LayerNorm((4096,), eps=1e-12, elementwise_affine=True)
              (attention): AlbertAttention(
                (query): Linear(in_features=4096, out_features=4096, bias=True)
                (key): Linear(in_features=4096, out_features=4096, bias=True)
                (value): Linear(in_features=4096, out_features=4096, bias=True)
                (attention_dropout): Dropout(p=0, inplace=False)
                (output_dropout): Dropout(p=0, inplace=False)
                (dense): Linear(in_features=4096, out_features=4096, bias=True)
                (LayerNorm): LayerNorm((4096,), eps=1e-12, elementwise_affine=True)
              )
              (ffn): Linear(in_features=4096, out_features=16384, bias=True)
              (ffn_output): Linear(in_features=16384, out_features=4096, bias=True)
              (activation): NewGELUActivation()
              (dropout): Dropout(p=0, inplace=False)
            )
          )
        )
      )
    )
  )
  (predictions): AlbertMLMHead(
    (LayerNorm): LayerNorm((128,), eps=1e-12, elementwise_affine=True)
    (dense): Linear(in_features=4096, out_features=128, bias=True)
    (decoder): Linear(in_features=128, out_features=30000, bias=True)
    (activation): NewGELUActivation()
  )
)
```

## Run
```
torchrun --nproc-per-node 4 pippy_albert.py
```

## Output
https://gist.github.com/kwen2501/3fd89d3f3f0c743d1e726c71c32a35e7

### CODE REVIEW

File: examples/hf/pippy_albert.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+import torch
+import torch.distributed as dist
+
+from pippy.IR import Pipe, PipeSplitWrapper, annotate_split_points
+from pippy.PipelineStage import PipelineStage
+
+from transformers import AlbertForMaskedLM, AlbertConfig
+
+from hf_utils import generate_inputs_for_model, get_number_of_params
```
Comment (lessw2020) [inline examples/hf/pippy_albert.py:17]: I can't locate the hf_utils file? 
I don't find it in the repo - am I missing an install or was this meant to be added?

Comment (lessw2020) [reply examples/hf/pippy_albert.py:17]: I found it - it's in another PR:
#875

File: examples/hf/pippy_albert.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    if args.cuda:
+        dev_id = args.rank % torch.cuda.device_count()
+        args.device = torch.device(f"cuda:{dev_id}")
+    else:
+        args.device = torch.device("cpu")
+
+    # Init process group
+    backend = "nccl" if args.cuda else "gloo"
+    dist.init_process_group(
```
Comment (lessw2020) [inline examples/hf/pippy_albert.py:106]: nit - should we start using device_mesh as defaults for nccl case?
(I did a quick mod and test and confirmed it works seamlessly for this example).  
I can add it as a PR on top though, so it is a sep discussion.

**Review by lessw2020 — APPROVED** [review]
looks good and works great! 
Thanks for adding this example.

File: examples/hf/pippy_albert.py (hunk 2 above)
Comment (kwen2501) [reply examples/hf/pippy_albert.py:106]: We can use either. Yeah, great if you can submit a PR! Thanks!

**Pull request merged**